### PR TITLE
Add Postgres SQL `DROP POLICY`

### LINKED ISSUE #15751: [OpenForce 2022] support parsing Postgre sql `DROP POLICY`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

## Background
ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). At present, this parser engine can handle SQLs for `MySQL`, `PostgreSQL`, `SQLServer`, `openGauss` and `Oracle`, which means we have to understand different database dialect SQLs.
More details: 
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

## Task
This issue is to solve some of the unsupported postgres sqls  in [this file](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/unsupported/unsupported.xml) .

- [ ] DROP POLICY

You can learn more details about this SQL [here](https://www.postgresql.org/docs/current/sql-commands.html).
- You may need to try to get why it's not supported.(antlr4 grammar? or not implement visit method) You can use [antlr4 plugins](https://plugins.jetbrains.com/plugin/7358-antlr-v4) to help you to analyze. You may need to visit [an official doc](https://www.postgresql.org/docs/current/sql-commands.html) to check the grammar.
- After you fix it, remember to add a new corresponding SQL case in [SQL Cases](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported) and expected parsed result in [Expected Statment XML](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/case).
- Run [SQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java) and [UnsupportedSQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/UnsupportedSQLParserParameterizedTest.java) to make sure no exceptions.
Notice, these issues can be a good example.
[support alter foreign table for pg/og ](https://github.com/apache/shardingsphere/pull/14944)
[support alter materialized view for pg/og.](https://github.com/apache/shardingsphere/pull/15331)

## Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr g4 file
3. Be familiar with Postgres SQLs
## Targets files
1. Postgres SQLs g4 file: https://github.com/apache/shardingsphere/blob/master/shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/org/apache/shardingsphere/sql/parser/autogen/PostgreSQLStatement.g4

### PR DESCRIPTION
**Pull request by everly-gif**
Fixes #15751 .

Changes proposed in this pull request:
- Adds `DROP POLICY`
- Added tests

### CODE REVIEW

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-policy.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-cases>
+    <sql-case id="drop-policy" value="DROP POLICY p1 ON my_table;" db-types="PostgreSQL" />
```
Comment (Liangda-w) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-policy.xml:20]: `sql-case id` doesn't match, pls use `drop_policy` consistently

**Review by Liangda-w — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/postgresql/ddl/PostgreSQLDropPolicyStatement.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import lombok.ToString;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DropPolicyStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.postgresql.PostgreSQLStatement;
+
+/**
+ * PostgreSQL drop policy statement.
+ */
+@ToString
+public final class PostgreSQLAlterGroupStatement extends DropPolicyStatement implements PostgreSQLStatement {
```
[the remark this answers is not in the document]
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/postgresql/ddl/PostgreSQLDropPolicyStatement.java]: Thank you ! Made the changes

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-policy.xml (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-policy.xml:20]: Made the changes

File: shardingsphere-sql-parser/shardingsphere-sql-parser-engine/src/main/java/org/apache/shardingsphere/sql/parser/core/database/visitor/SQLVisitorRule.java [hunk 3]
```diff
@@ -431,6 +431,8 @@
     ALTER_TEXT_SEARCH_TEMPLATE("AlterTextSearchTemplate", SQLStatementType.DDL),
     
     ALTER_TEXT_SEARCH_PARSER("AlterTextSearchParser", SQLStatementType.DDL);
+    
+    DROP_POLICY("DropPolicy", SQLStatementType.DDL),
```
Comment (Liangda-w) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-engine/src/main/java/org/apache/shardingsphere/sql/parser/core/database/visitor/SQLVisitorRule.java:435]: This line should end with a semicolon, and line 433 with a comma

**Review by Liangda-w — CHANGES_REQUESTED** [review]
Please always make sure the ci runs properly, you can run `maven clean install -Drat.skip=true -Dmaven.javadoc.skip=true` locally before the commit

**Comment by Liangda-w** [comment]
You still have a Continous Integration error, please check it and fix it by yourself

image

**Comment by Liangda-w** [comment]
@everly-gif Please also delete the now supported drop policy SQLs in `shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/unsupported/unsupported.xml`

**Review by Liangda-w — APPROVED** [review]
👍

**Pull request merged**